src/Analyzers/VisualBasic/Analyzers/IncludeParenthesesForMethodInvocations/VisualBasicIncludeParenthesesForMethodInvocationsDiagnosticAnalyzer.vb

```diff
@@ -0,0 +1,64 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Imports Microsoft.CodeAnalysis.Diagnostics
+Imports Microsoft.CodeAnalysis.VisualBasic.Precedence
+Imports Microsoft.CodeAnalysis.VisualBasic.LanguageServices
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+Imports Microsoft.CodeAnalysis.RemoveUnnecessaryParentheses
+Imports Microsoft.CodeAnalysis.Precedence
+Imports Microsoft.CodeAnalysis.LanguageServices
+Imports Microsoft.CodeAnalysis.CodeStyle
+Imports System.Collections.Immutable
+Imports Microsoft.CodeAnalysis.Options
+Imports Microsoft.CodeAnalysis.VisualBasic.CodeStyle
+Imports Microsoft.CodeAnalysis.Operations
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.IncludeParenthesesForMethodInvocations
+    <DiagnosticAnalyzer(LanguageNames.VisualBasic)>
+    Friend NotInheritable Class VisualBasicIncludeParenthesesForMethodInvocationsDiagnosticAnalyzer
+        Inherits AbstractBuiltInCodeStyleDiagnosticAnalyzer
+
+        Public Sub New()
+            MyBase.New(
+                diagnosticId:=IDEDiagnosticIds.IncludeParenthesesForMethodInvocationsDiagnosticId,
+                [option]:=VisualBasicCodeStyleOptions.IncludeParenthesesForMethodInvocations,
+                language:=LanguageNames.VisualBasic,
+                title:=New LocalizableResourceString(NameOf(VisualBasicAnalyzersResources.Add_invocation_parentheses), VisualBasicAnalyzersResources.ResourceManager, GetType(VisualBasicAnalyzersResources)))
+        End Sub
+
+        Protected Overrides Sub InitializeWorker(context As AnalysisContext)
+            context.RegisterOperationAction(AddressOf AnalyzeInvocations, OperationKind.Invocation)
+        End Sub
+
+        Private Sub AnalyzeInvocations(context As OperationAnalysisContext)
+            Dim includeParen As Boolean = context.Options.GetOption(VisualBasicCodeStyleOptions.IncludeParenthesesForMethodInvocations, context.Operation.Syntax.SyntaxTree, context.CancellationToken).Value
+            Dim node = context.Operation.Syntax
+            Dim operation = DirectCast(context.Operation, IInvocationOperation)
+            If TypeOf node IsNot InvocationExpressionSyntax Then
+                ' This should never happen I think?
+                ' Should it be removed or kept as defensive check?
+                Return
+            End If
```

Comment:
you absolutely need this check.  it is not certain that IOperations will be backed by any particular syntax.